File: evidence/generated-systems/angular/governance-on-angular/src/app/models/RetentionSchedule.js

// Define collection and schema for RetentionSchedule
export interface RetentionSchedule {
    name:
	type : string
    retentionPeriodMonths:
	type : number
    Repositories:
 	type : [{ type: Schema.Types.ObjectId, ref: 'RecordsRepository' }]
    Records:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Record_' }]
    Exceptions:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Exception_' }]
    DispositionReviews:
 	type : [{ type: Schema.Types.ObjectId, ref: 'DispositionReview' }]
    RetentionTrigger:
 	type : String
    DispositionAction:
 	type : String
    Status:
 	type : String
#
    collection: 'retentionSchedules'
}

File: evidence/generated-systems/golang/governance-on-golang/governance-on-golang/internal/model/RetentionSchedule.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// RetentionSchedule Declaration
//==============================================================
type RetentionSchedule struct {
    gorm.Model
     Name                                    string
    RetentionPeriodMonths                                                            string
     Repositories           []RecordsRepository `gorm:"foreignKey:RepositoriesFromRetentionScheduleId"`
     Records           []Record_ `gorm:"foreignKey:RecordsFromRetentionScheduleId"`
     Exceptions           []Exception_ `gorm:"foreignKey:ExceptionsFromRetentionScheduleId"`
     DispositionReviews           []DispositionReview `gorm:"foreignKey:DispositionReviewsFromRetentionScheduleId"`
    RetentionTrigger                      RetentionTrigger
    DispositionAction                      DispositionAction
    Status                      RetentionStatus

// parent associations as their child

}

